src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -0,0 +1,244 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Comment } from '../../types/Comment';
+import { postComments } from '../../utils/commentsApi';
+
+type Props = {
+  postId: number;
+  onAddComment: (comment: Comment) => void;
+};
+
+const fieldStatus = {
+  hasName: true,
+  hasEmail: true,
+  hasComment: true,
+};
+
+export const NewCommentForm: React.FC<Props> = ({ postId, onAddComment }) => {
+  const [nameValue, setNameValue] = useState('');
+  const [emailValue, setEmailValue] = useState('');
+  const [commentValue, setCommentValue] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [hasInputValue, setHasInputValue] = useState(fieldStatus);
```

Comment:
redundant state actually)

you can check commentValue?.length right away at all places you need instead of hasInputValue.hasComment

Reply:
it will decrease a number of code

Reply:
I understand, but if I do this, then when I write a new comment, all the fields turn red, and only after entering it disappears. Because, in the beginning, I don't have any value

Reply:
ok, let it be, in future you will use react-hook-form or formik which are will make it much easier